spec/plugin_spec.rb

```diff
@@ -195,6 +195,15 @@
     expect(result).to eq nil
   end
 
+  it 'can walk json and find values by index in an array' do
+    authenticator = OAuth2BasicAuthenticator.new
+    json_string = '{"emails":[{"value":"test@example.com"},{"value":"test2@example.com"}]}'
+    SiteSetting.oauth2_json_email_path = 'emails[1].value'
+    result = authenticator.json_walk({}, JSON.parse(json_string), :user_id)
+
+    expect(result).to eq nil
```

Comment:
shouldn't we be expecting `test2@example.com` here?

Reply:
Thanks! Had a copy issue and because of how I was sort of pre-testing the tests locally the issue with it incorrectly passing for me didn't show up. The passed in token also should have been :email, or the oauth2_json path should have been the oauth2_json_user_id_path